compiler/main.go

```diff
@@ -28,27 +30,53 @@ var sourceFileName = map[string]string{
 }
 
 func main() {
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+	consoleWriter := zerolog.ConsoleWriter{Out: os.Stdout}
+	logFile, logErr := os.OpenFile("compiler.log", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0600)
+	defer logFile.Close()
+	if logErr != nil {
+		log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
+		log.Fatal().Err(logErr).Msg("Could not create log file")
+	}
+	multi := zerolog.MultiLevelWriter(consoleWriter, logFile)
+	logger := zerolog.
+					New(multi).
+					With().
+					Timestamp().
+					Str("MaRT", "compiler").
+					Logger().
+					Level(zerolog.DebugLevel)
+
 	m := martini.Classic()
 	m.Post("/", binding.MultipartForm(FileForm{}), func(rs http.ResponseWriter, rq *http.Request, req FileForm) {
 		fileName := sourceFileName[req.Language]
 		f, createErr := os.Create(fileName)
 		if createErr != nil {
 			msg := "An error occurred while trying to create a file named '" + fileName + "'"
+			logger.Error().
+					Err(createErr).
+					Msg(msg)
 			errors.WriteResponse(rs, http.StatusBadRequest, msg, createErr)
 			return
 		}
 		program, pErr := req.Program.Open()
 		if pErr != nil {
 			msg := "An error occurred while trying to open the received program"
+			logger.Error().
```

Comment:
Sometimes you break this into three lines and sometimes (in other files) you don't, but it seems to be the same thing. Even the size

Reply:
Have go fmt run